expr/arithmetic_test.go

```diff
@@ -3,74 +3,9 @@ package expr_test
 import (
 	"testing"
 
-	"github.com/genjidb/genji/document"
-	"github.com/genjidb/genji/expr"
+	"github.com/genjidb/genji/testutil"
 )
 
-func TestArithmeticExpr(t *testing.T) {
-	tests := []struct {
-		expr  string
-		res   document.Value
-		fails bool
-	}{
-		{"1 + a", document.NewIntegerValue(2), false},
-		{"1 + NULL", nullLitteral, false},
-		{"1 + notFound", nullLitteral, false},
-		{"1 - a", document.NewIntegerValue(0), false},
-		{"1 - NULL", nullLitteral, false},
-		{"1 - notFound", nullLitteral, false},
-		{"1 * a", document.NewIntegerValue(1), false},
-		{"1 * NULL", nullLitteral, false},
-		{"1 * notFound", nullLitteral, false},
-		{"1 / a", document.NewIntegerValue(1), false},
-		{"1 / NULL", nullLitteral, false},
-		{"1 / notFound", nullLitteral, false},
-		{"1 % a", document.NewIntegerValue(0), false},
-		{"1 % NULL", nullLitteral, false},
-		{"1 % notFound", nullLitteral, false},
-		{"1 & a", document.NewIntegerValue(1), false},
-		{"1 & NULL", nullLitteral, false},
-		{"1 & notFound", nullLitteral, false},
-		{"1 | a", document.NewIntegerValue(1), false},
-		{"1 | NULL", nullLitteral, false},
-		{"1 | notFound", nullLitteral, false},
-		{"1 ^ a", document.NewIntegerValue(0), false},
-		{"1 ^ NULL", nullLitteral, false},
-		{"1 ^ notFound", nullLitteral, false},
-	}
-
-	for _, test := range tests {
-		t.Run(test.expr, func(t *testing.T) {
-			testExpr(t, test.expr, envWithDoc, test.res, test.fails)
-		})
-	}
-}
-
-func TestArithmeticExprNodocument(t *testing.T) {
-	tests := []struct {
-		expr  string
-		res   document.Value
-		fails bool
-	}{
-		{"1 + a", nullLitteral, true},
-		{"1 - a", nullLitteral, true},
-		{"1 * a", nullLitteral, true},
-		{"1 / a", nullLitteral, true},
-		{"1 % a", nullLitteral, true},
-		{"1 & a", nullLitteral, true},
-		{"1 | a", nullLitteral, true},
-		{"1 ^ a", nullLitteral, true},
-	}
-
-	for _, test := range tests {
-		t.Run(test.expr, func(t *testing.T) {
-			for _, test := range tests {
-				t.Run(test.expr, func(t *testing.T) {
-					emptyenv := expr.NewEnvironment(nil)
-
-					testExpr(t, test.expr, emptyenv, test.res, test.fails)
-				})
-			}
-		})
-	}
+func TestArithmetic(t *testing.T) {
+	testutil.ExprRunner(t, "testdata/arithmetic.sql")
```

Comment:
This will probably fail on Windows unless I’ve missed filepath.FromSlash somewhere. We should either call filepath.Join here, or use filepath.FromSlash in ExprRunner. Alternatively, we can use Open on fs.FS that is returned by os.DirFS instead of os.Open.

Reply:
Good catch, thanks!

Reply:
I think it’s better to keep paths consistent across platforms, so actually let’s use either FromSlash or fs.FS.